classes/hook_callbacks.php

```diff
@@ -38,4 +39,27 @@ public static function before_standard_top_of_body_html_generation(before_standa
         // Get code to inject.
         $hook->add_html(envbarlib::get_inject_code());
     }
+
+    /**
+     * Listener for the after_config hook.
+     *
+     * @param \core\hook\after_config $hook
+     */
+    public static function after_config(\core\hook\after_config $hook): void {
+        global $CFG;
+
+        if (during_initial_install() || isset($CFG->upgraderunning)) {
+            // Do nothing during installation or upgrade.
+            return;
+        }
+
+        // Hack to avoid breaking messaging tests, as this setting defaults on.
```

Comment:
Can you please provide a bit more details on what issue are you talking about? What setting is on by default?

Reply:
Ah I see you just copied it from lib.php

Reply:
Do we need just call local_envbar_after_config() in this hook to avoid the code duplication?

Reply:
Hi @dmitriim ,

I have linked to an issue and update the hook.